features/device-config/service/src/main/java/org/opennms/features/deviceconfig/service/impl/DeviceConfigServiceImpl.java

```diff
@@ -81,50 +97,98 @@ public void triggerConfigBackup(String ipAddress, String location, String config
+                    return new SimpleMonitoredService(ipInterface.getIpAddress(), node.getId(), node.getLabel(), optional.get().getServiceName(), location);
+                }
+            }
             return new SimpleMonitoredService(ipInterface.getIpAddress(), node.getId(), node.getLabel(), serviceName, location);
         });
 
         if (service == null) {
             throw new IllegalArgumentException("No interface found with ipAddress " + ipAddress + " at location " + location);
         }
+        List<Parameter> parameters = getServiceParamsFromPollerConfig(service);
+        Map<String, Object> serviceAttributes = convertParamsToAttributes(parameters);
         // All the service parameters should be loaded from metadata in PollerRequestBuilderImpl
         // Persistence will be performed in DeviceConfigMonitorAdaptor.
         return locationAwarePollerClient.poll()
                 .withService(service)
                 .withAdaptor(serviceMonitorAdaptor)
                 .withMonitorClassName(DEVICE_CONFIG_MONITOR_CLASS_NAME)
+                .withAttributes(serviceAttributes)
                 .execute();
     }
 
+    private List<Parameter> getServiceParamsFromPollerConfig(MonitoredService service) throws IOException {
+        final PollerConfig pollerConfig = ReadOnlyPollerConfigManager.create();
+        List<OnmsMetaData> metaData = sessionUtils.withReadOnlyTransaction(() -> {
+            OnmsMonitoredService monitoredService = monitoredServiceDao.get(service.getNodeId(), service.getAddress(), service.getSvcName());
+            List<OnmsMetaData> metaDataList = monitoredService.getMetaData();
+            return metaDataList.stream().filter(onmsMetaData -> onmsMetaData.getContext().equals("requisition")).collect(Collectors.toList());
+        });
+        Optional<OnmsMetaData> pkgFromMetadata = metaData.stream().filter(onmsMetaData -> onmsMetaData.getKey().equals("package")).findFirst();
+        List<Parameter> parameters = new ArrayList<>();
+        if(pkgFromMetadata.isPresent()) {
+            org.opennms.netmgt.config.poller.Package pkg = pollerConfig.getPackage(pkgFromMetadata.get().getValue());
+            parameters.addAll(pkg.getService(service.getSvcName()).getParameters());
+        } else {
+            List<String> pkgNames = pollerConfig.getAllPackageMatches(service.getIpAddr());
+            List<org.opennms.netmgt.config.poller.Package> packages =
+                    pollerConfig.getPackages().stream().filter(pollerPackage ->
+                            pollerConfig.isServiceInPackageAndEnabled(service.getSvcName(), pollerPackage)).collect(Collectors.toList());
+            Optional<org.opennms.netmgt.config.poller.Package> pkg = packages.stream().filter(pollerPackage ->
+                    pkgNames.contains(pollerPackage.getName())).findFirst();
+            pkg.ifPresent(aPackage -> parameters.addAll(aPackage.getService(service.getSvcName()).getParameters()));
+        }
```

Comment:
I think, having the packagae defined in metadata is a bad idea. This is super hard to debug and works differently to what the real poller does. We should strictly mimic the behavior of `Poller::findPackageForService` (ideally by moving this to the `pollerConfig`),

Reply:
This is only needed in very limited scenario where there are multiple packages with  `DeviceConfig` service that have overlapping IPAddresses that belong to different locations. There is no easy way to determine package in this case.

Reply:
Since overlapping IpAddresses is not really handled in poller packages, removed code to retrieve package from metadata.